src/coreclr/debug/daccess/CMakeLists.txt

```diff
@@ -47,10 +47,10 @@ target_link_libraries(daccess PRIVATE cdacreader_api)
 
 add_dependencies(daccess eventing_headers)
 
-if(CLR_CMAKE_HOST_FREEBSD OR CLR_CMAKE_HOST_NETBSD OR CLR_CMAKE_HOST_SUNOS)
+if(CLR_CMAKE_HOST_FREEBSD OR CLR_CMAKE_HOST_NETBSD OR CLR_CMAKE_HOST_SUNOS OR CLR_CMAKE_HOST_APPLE)
```

Comment:
@hoyosjs, I think it was this change that broke debugging/diagnostics repo tests on MacOS x64. Adding CLR_CMAKE_HOST_APPLE here will cause the runtime DAC table to be generated in the old "rva" way.

Reply:
I don't think the change was intentional. I traced it back to the first commit on my previous branch but it could have been a result of some rebase.

Reply:
Do you want to remove/fix this?

Reply:
Seems like at some point after the initial patch the conditional compilation of machoreader.cpp was added. This still uses the old `CLR_CMAKE_TARGET_OSX` condition. I'll send a PR to fix this.

Reply:
https://github.com/dotnet/runtime/pull/112076

Reply:
> @hoyosjs, I think it was this change that broke debugging/diagnostics repo tests on MacOS x64. Adding CLR_CMAKE_HOST_APPLE here will cause the runtime DAC table to be generated in the old "rva" way.

Thanks Mike for drawing attention to this! 

Which pipeline is running these tests? 
I am thinking about ways to improve test coverage for PRs like this, so that we can catch such regressions earlier.

Reply:
These failing tests are part of the diagnostics repo.  There is a manual way of running them against a local runtime build, but there currently no way for the runtime repo to use them.  There is Jeremy created this issue https://github.com/dotnet/diagnostics/issues/5213 to track this work.

Reply:
Thanks a lot for the pointers.